sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -36,4 +56,145 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
+      offering.programRelatedCosts =
+        scholasticStandingData.booksAndSupplies ??
+        existingOffering.programRelatedCosts;
+      offering.mandatoryFees =
+        scholasticStandingData.mandatoryFees ?? existingOffering.mandatoryFees;
+      offering.exceptionalExpenses =
+        scholasticStandingData.exceptionalCosts ??
+        existingOffering.exceptionalExpenses;
+
+      offering.tuitionRemittanceRequestedAmount =
+        existingOffering.tuitionRemittanceRequestedAmount;
+      offering.offeringDelivered = existingOffering.offeringDelivered;
+      offering.lacksStudyDates = existingOffering.lacksStudyDates;
+      offering.lacksStudyBreaks = existingOffering.lacksStudyBreaks;
+      offering.lacksFixedCosts = existingOffering.lacksFixedCosts;
+      offering.tuitionRemittanceRequested =
+        existingOffering.tuitionRemittanceRequested;
+      offering.educationProgram = {
+        id: existingOffering.educationProgram.id,
+      } as EducationProgram;
+      offering.institutionLocation = {
+        id: existingOffering.institutionLocation.id,
+      } as InstitutionLocation;
+
+      // TODO: When new offering type is added for report a change/ scholastic standing, update this
+      offering.offeringType = OfferingTypes.applicationSpecific;
+
+      offering.offeringIntensity = existingOffering.offeringIntensity;
+      offering.yearOfStudy = existingOffering.yearOfStudy;
+      offering.showYearOfStudy = existingOffering.showYearOfStudy;
+      offering.hasOfferingWILComponent =
+        existingOffering.hasOfferingWILComponent;
+      offering.offeringWILType = existingOffering.offeringWILType;
+      offering.studyBreaks = existingOffering.studyBreaks;
+      offering.offeringDeclaration = existingOffering.offeringDeclaration;
+
+      // Save new offering.
+      await transactionalEntityManager
+        .getRepository(EducationProgramOffering)
+        .save(offering);
+
+      // Create StudentScholasticStanding.
+      const scholasticStanding = new StudentScholasticStanding();
+      scholasticStanding.application = { id: applicationId } as Application;
+      scholasticStanding.submittedData = scholasticStandingData;
+      scholasticStanding.submittedDate = now;
+      scholasticStanding.scholasticStandingStatus =
+        ScholasticStandingStatus.Approved;
+      scholasticStanding.submittedBy = auditUser;
+      scholasticStanding.creator = auditUser;
+
+      // Create the new assessment to be processed.
+      scholasticStanding.studentAssessment = {
+        application: { id: applicationId } as Application,
+        triggerType: AssessmentTriggerType.ScholasticStandingChange,
+        creator: auditUser,
+        submittedBy: auditUser,
+        submittedDate: now,
+        offering: { id: offering.id } as EducationProgramOffering,
+      } as StudentAssessment;
+
+      const studentScholasticStandingObj = await transactionalEntityManager
```

Comment:
Please avoid suffixes like "Obj". I do not believe that we are using it in general.